src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -1116,11 +1117,18 @@ public void AcceptNewConnection(Stream stream, Action disconnect, EndPoint remot
             }
         }
 
-        private static void ValidateCompatibility(RachisHello initialMessage)
+        private void ValidateCompatibility(RachisHello initialMessage, ClusterTopology clusterTopology)
         {
             var version = initialMessage.ServerBuildVersion;
-            if (version == 0 || ServerVersion.IsNightlyOrDev(version))  
+            if (ServerVersion.IsNightlyOrDev(version))
                 return;
+
+            if (clusterTopology.Members.ContainsKey(Tag) == false)
+            {
+                if (version == 0 || (version >= 40_000 && version < 42_000))
```

Comment:
The problem with version 0 is that we can't distinguish between 4.1.x and 4.2.0 :/